src/bpmncwpverify/visitors.py

```diff
@@ -465,3 +450,391 @@ def visit_bpmn(self, bpmn: Bpmn) -> None:
     def end_visit_bpmn(self, bpmn: Bpmn) -> None:
         for place in self.flow_places:
             self.write_places_lines("bit {x} = 0".format(x=str(place)))
+
+
+class CwpGraphVizVisitor(CwpVisitor):  # type: ignore
+    def __init__(self) -> None:
+        self.dot = graphviz.Digraph(comment="cwp graph")
+
+    def visit_state(self, state: CwpState) -> None:
+        self.dot.node(state.name, state.name)
+
+    def end_visit_state(self, state: CwpState) -> None:
+        pass
+
+    def visit_edge(self, edge: CwpEdge) -> None:
+        if edge.source:
+            self.dot.edge(edge.source.name, edge.dest.name, label=edge.expression)
+        else:
+            self.dot.edge("start", edge.dest.name, label=edge.expression)
+
+    def end_visit_edge(self, edge: CwpEdge) -> None:
+        pass
+
+    def visit_cwp(self, model: Cwp) -> None:
+        pass
+
+    def end_visit_cwp(self, model: Cwp) -> None:
+        pass
+
+
+class CwpLtlVisitor(CwpVisitor):  # type: ignore
+    def __init__(self, symbol_table: SymbolTable, print_on: bool = False) -> None:
+        self.state_info: List[str] = []
+        self.symbol_table = symbol_table
+        self.output_str: List[str] = []
+        self.print_on: bool = print_on
+        self.property_list: List[str] = []
+        self.cwp: Cwp
+        self.debug: bool = False
+        self.tab = 0
+
+    def generate_all(self) -> None:
+        self.generate_helper_functions()
+        self.generate_LTL()
+
+    def generate_helper_functions(self) -> None:
+        self.write_line("")
+        self.write_line("")
+        self.write_state_variables()
+        self.write_line("")
+        self.write_line("")
+        self.write_variable_range_invariants()
+        self.write_line("")
+        self.write_line("")
+        self.write_init_states()
+        self.write_line("")
+        self.write_line("")
+        self.write_edge_definitions()
+        self.write_line("")
+        self.write_line("")
+        self.write_update_state()
+        self.write_line("")
+        self.write_line("")
+        if self.print_on:
+            self.write_log_state_inline()
+            self.write_line("")
+            self.write_line("")
+
+    def generate_LTL(self) -> None:
```

Comment:
Each of these `generate` functions should have a test make sure in isolation they behave as expected. That may be more work than merit for some of these, but where possible, it would be nice to generate test inputs to test the functionality of each of these in isolation.